api/src/api/members.js

```diff
@@ -72,6 +72,74 @@ router.post(
   }),
 );
 
+// Gives the enum options to the frontend to populate the dropdown.
+// The key in the 'options' object must be the same as the DB attribute.
+router.get(
+  '/options',
+  requireRegistered,
+  errorWrap(async (req, res) => {
+    const options = {
+      level: Member.levelEnum,
+      location: Member.locationEnum,
+      role: Member.roleEnum,
+      gradSemester: Member.semesterEnum,
+      generationSemester: Member.semesterEnum,
+      classStanding: Member.classStandingEnum,
+      status: Member.statusEnum,
+    };
```

Comment:
Rather than explicitly specifying which fields have options & their corresponding options, I'd suggest
```es6
const options = {};
Member.schema.eachPath((pathname, schemaType) => {
  const { enum: optionsEnum } = schemaType.options;
  if (optionsEnum) {
    options[pathname] = optionsEnum;
  }
});
```
Also `options[level]` would be an array of options now. `options`:
```es6
{
  gradSemester: [ 'FALL', 'SPRING' ],
  classStanding: [
    'FRESHMAN',
    'SOPHOMORE',
    'JUNIOR',
    'SPRING_SENIOR',
    'FALL_SENIOR',
    'MASTERS',
    'PHD'
  ],
  generationSemester: [ 'FALL', 'SPRING' ],
  location: [ 'REMOTE', 'CAMPUS', 'TBD' ],
  role: [
    'CO_DIRECTOR',
    'COMMUNITY_DIRECTOR',
    'EXTERNAL_DIRECTOR',
    'ACADEMY_LEAD',
    'TECH_LEAD',
    'PR_LEAD',
    'PRODUCT_MANAGER',
    'PRODUCT_DESIGNER',
    'DEVELOPER',
    'ACADEMY_MEMBER'
  ],
  level: [ 'ADMIN', 'DIRECTOR', 'LEAD', 'MEMBER', 'TBD' ],
  status: [ 'ACTIVE', 'HIATUS', 'RETIRED', 'TBD' ]
}
```

Reply:
You could even move this functionality (generating the options object) to `Member` to keep the endpoint itself more barebones.

Reply:
I'll replace the first part of the endpoint with that, but I'm not sure about changing the format of the response.

The dropdown expects each item to have a label and value field, so that processing needs to be done on the frontend or the backend. I decided to push it to the backend because I wanted the frontend to do as little processing as possible, and I think we're only going to get these fields in the context of dropdowns. But I'd like to hear your opinions on this.

Reply:
I think both ways make sense! In the current response format it would allow the backend to specify the label as well, which is more flexible.